ttnn/cpp/ttnn/tensor/tensor_layout.cpp

```diff
@@ -0,0 +1,358 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "tensor_layout.hpp"
+#include "ttnn/tensor/enum_types.hpp"
+#include "types.hpp"
+
+namespace tt::tt_metal {
+
+namespace {
+size_t round_up(size_t value, size_t multiple) {
+    // can be faster if multiple is power of 2
+    // return (value + multiple - 1) & ~(multiple - 1);
+    return ((value + multiple - 1) / multiple) * multiple;
+};
+}
+
+Size::Size(size_t height, size_t width) : mHeight(height), mWidth(width) {}
+Size::Size(const std::pair<size_t, size_t>& size) : mHeight(size.first), mWidth(size.second) {}
+Size::Size(const std::array<size_t, 2>& size) : mHeight(size[0]), mWidth(size[1]) {}
+
+
+Size Size::operator/(const Size& rhs) const {
+    return Size(mHeight / rhs.mHeight, mWidth / rhs.mWidth);
+}
+
+Size Size::operator*(size_t scalar) const {
+    return Size(mHeight * scalar, mWidth * scalar);
+}
+
+Size::operator std::pair<size_t, size_t>() const {
+    return {mHeight, mWidth};
+}
+
+Size::operator std::array<size_t, 2>() const {
+    return {mHeight, mWidth};
+}
+
+Size::operator std::array<uint32_t, 2>() const {
+    return {static_cast<uint32_t>(mHeight), static_cast<uint32_t>(mWidth)};
+}
+
+size_t Size::height() const { return mHeight; }
+size_t Size::width() const { return mWidth; }
+
+bool Size::operator==(const Size& rhs) const {
+    return mHeight == rhs.mHeight && mWidth == rhs.mWidth;
+}
+
+// does not have to be a member, but it is easier to find if it is
+Size Size::aligned_to_tile(const Size& tile) {
+    size_t height = round_up(mHeight, tile.height());
+    size_t width = round_up(mWidth, tile.width());
+    return Size(height, width);
+}
+
+std::ostream& operator<<(std::ostream& os, const tt::tt_metal::Size& size)
+{
```

Comment:
Please put `{` on the previous line to be consistent with formatting

Reply:
really need to get clang-format in.. :D 
thanks!